src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -350,12 +350,18 @@ def is_ceph_disk_member(self):
 
     @property
     def has_bluestore_label(self):
-        out, err, ret = process.call([
-            'ceph-bluestore-tool', 'show-label',
-            '--dev', self.abspath], verbose_on_failure=False)
-        if ret:
-            return False
-        return True
+        fd = None
+        isBluestore = False
+        bluestoreDiskSignature = 'bluestore block device' # 22 bytes long
+
+        # throws OSError on failure
+        with os.open(self.abspath, os.O_RDONLY, 0) as fd:
+            # read first 22 bytes looking for bluestore disk signature
+            signature = os.read(fd, 22)
+            if signature.decode('ascii', 'replace') == bluestoreDiskSignature:
+                isBluestore = True
```

Comment:
You still need to close the fd otherwise it's leaking.

Reply:
You don't need to close the file when using the `with` keyword.

This is done automatically even if an error is raised.